### PR TITLE
Adding process.name field and minor enhancements to od* events.

### PR DESCRIPTION
**Pull request by txhaflaire**
Type of change:
- Enhancement

## Proposed commit message

- Added `process.name` field, this for preparations for Event Visualiser
- Minor enhancements to `od*` pipeline events
  - `pipeline_event_od_group_add` -> added `event.type` field
  - `pipeline_event_od_group_remove` -> added `event.type` field
 - Minor enhancements to `screensharing*` events
    - `pipeline_event_screensharing_attach` added `jamf_protect.telemetry.source_address_type` field
    - `pipeline_event_screensharing_detach` added `jamf_protect.telemetry.source_address_type` field

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally
`elastic-package test system`

```
2024/08/19 16:45:26  INFO Write container logs to file: /Users/thijs.xhaflaire/Documents/GitHub/Elastic/integrations/build/container-logs/jamf-protect-webtraffic-http-endpoint-1724078726146769000.log
--- Test results for package: jamf_protect - START ---
╭──────────────┬────────────────────┬───────────┬───────────────┬────────┬───────────────╮
│ PACKAGE      │ DATA STREAM        │ TEST TYPE │ TEST NAME     │ RESULT │  TIME ELAPSED │
├──────────────┼────────────────────┼───────────┼───────────────┼────────┼───────────────┤
│ jamf_protect │ alerts             │ system    │ http-endpoint │ PASS   │ 56.502828708s │
│ jamf_protect │ telemetry          │ system    │ http-endpoint │ PASS   │ 52.955471459s │
│ jamf_protect │ telemetry_legacy   │ system    │ http-endpoint │ PASS   │  57.11250125s │
│ jamf_protect │ web_threat_events  │ system    │ http-endpoint │ PASS   │      55.1454s │
│ jamf_protect │ web_traffic_events │ system    │ http-endpoint │ PASS   │ 52.103597834s │
╰──────────────┴────────────────────┴───────────┴───────────────┴────────┴───────────────╯
--- Test results for package: jamf_protect - END   ---
Done
```

### CODE REVIEW

**Comment by txhaflaire** [comment]
@kcreddy Please another `/test` - i've fixed the mismatching files (readme/sample_event.json)

File: packages/jamf_protect/data_stream/telemetry/elasticsearch/ingest_pipeline/pipeline_event_authentication.yml [hunk 1]
```diff
@@ -405,6 +405,18 @@ processors:
...
+        lang: painless
+        source: >
+            if (ctx.process?.executable != null) {
+            int lastSlashIndex = ctx.process.executable.lastIndexOf('/');
+            if (lastSlashIndex != -1) {
+                ctx.process.name = ctx.process.executable.substring(lastSlashIndex + 1);
+            } else {
+                ctx.process.name = ctx.process.executable; // Fallback if no slash is found
+            }
+            }
```
Comment (kcreddy) [inline packages/jamf_protect/data_stream/telemetry/elasticsearch/ingest_pipeline/pipeline_event_authentication.yml:418]:
```suggestion
        source: >
            if (ctx.process?.executable != null) {
                int lastSlashIndex = ctx.process.executable.lastIndexOf('/');
                if (lastSlashIndex != -1) {
                    ctx.process.name = ctx.process.executable.substring(lastSlashIndex + 1);
                } else {
                    ctx.process.name = ctx.process.executable; // Fallback if no slash is found
                }
            }
```

Comment (kcreddy) [reply packages/jamf_protect/data_stream/telemetry/elasticsearch/ingest_pipeline/pipeline_event_authentication.yml:418]: Could you indent this slightly better? Thanks!

File: packages/jamf_protect/changelog.yml [hunk 2]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "2.4.0"
+  changes:
+    - description: Added process.name and some minor enhancements to some events
+      type: bugfix
```
Comment (kcreddy) [inline packages/jamf_protect/changelog.yml:5]:
```suggestion
      type: enhancement
```

File: packages/jamf_protect/changelog.yml [hunk 3]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "2.4.0"
+  changes:
+    - description: Added process.name and some minor enhancements to some events
```
Comment (kcreddy) [inline packages/jamf_protect/changelog.yml:4]: Could you please add what these minor enhancements are in the PR description's commit message section?

Comment (txhaflaire) [reply packages/jamf_protect/changelog.yml:4]: Done

**Comment by txhaflaire** [comment]
@kcreddy Thanks for the review and the suggestions, suggestions have been implemented.

**Review by kcreddy — APPROVED** [review]

**Pull request merged**